src/main/java/com/google/sps/servlets/ListOrganizationsServlet.java

```diff
@@ -92,10 +71,47 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(requestedOrganizations));
   }
+  /* Given a user id, this function checks if that user has a maintainer role in the datastore */
+  public boolean userIsMaintainer(String userId) {
+    //TODO(): Check if given user ID has maintainer role in the datastore
+    return true;
+  }
+
+  /* This function constructs a query based on the request parameters & user's role */
+  public Query getQueryFromParams(HttpServletRequest request, UserService userService) {
+    Query query = new Query("Distributor").addSort("creationTimeStamp", SortDirection.DESCENDING);;
+
+    /* displayUserOrgsParameter is true when user only wants to see orgs they moderate*/
+    String displayUserOrgsParameter = request.getParameter("displayUserOrgs");
+    boolean displayUserOrgs = userOrgsParameterHelper(request);
+    boolean isUserLoggedIn = userService.isUserLoggedIn();
+    String userId = userService.getCurrentUser().getUserId();
+    boolean userIsMaintainer = userIsMaintainer(userId);
+
+    if (isUserLoggedIn && displayUserOrgs) {
+      /* If the user is logged in and wants to just see their orgs, get their user ID & index with it*/
+      query.setFilter(new FilterPredicate("moderatorList", FilterOperator.EQUAL, userId));
+    }
+
+    if (!userIsMaintainer) {
+      /* If the user is not a maintainer, only allow them to see approved orgs */
+      query.setFilter(new FilterPredicate("isApproved", FilterOperator.EQUAL, true));
+    }
+
+    // TODO(): Read through request parameters and for all valid parameters and use them to modify query (filtering)
 
-  public Query ConstructQueryForUserInfo(String userID) {
-    Query query = new Query("Distributor").setFilter(new FilterPredicate("moderatorList",
-                    FilterOperator.EQUAL, userID)).addSort("creationTimeStampMillis", SortDirection.DESCENDING);
     return query;
   }
+
+  public boolean userOrgsParameterHelper(HttpServletRequest request) {
```

Comment:
suggestion: pass in a String key here.

now the function can be called:

coerceParameterToBoolean(request, key) {
  // all your code is the same, but instead of getParameter("displayUserOrgs") you do getParameter(key)
}